edi_utm_oca/models/edi_exchange_consumer_mixin.py

```diff
@@ -0,0 +1,33 @@
+# Copyright 2024 Camptocamp
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+
+from odoo import api, models
+
+
+class EDIExchangeConsumerMixin(models.AbstractModel):
+    _inherit = "edi.exchange.consumer.mixin"
+
+    def _edi_update_source(self, vals):
+        conditions = [
+            vals.get("origin_exchange_record_id"),
+            "source_id" not in vals,
+            hasattr(self, "source_id"),
+        ]
+
+        if all(conditions):
+            utm_source_edi = self.env.ref(
+                "edi_utm_oca.utm_source_edi",
+                raise_if_not_found=False,
+            )
```

Comment:
```suggestion
            utm_source_edi = self._edi_get_utm_source(vals)
```
to offer the hook for customization